util/stream.go

```diff
@@ -143,17 +151,21 @@ func (m *MessageStream) inbound() {
 
 // Parse incoming message
 func (m *MessageStream) parse() {
+	errMessage := "received: %v and encountered error: %v"
 	for {
-		b := <-m.pool.Full
-		log.Debugf("Rcvd: %v", b.Bytes())
-		msg, err := m.parser.Parse(b.Bytes())
-		// Log all message parsing errors.
-		if err != nil {
-			log.Print(err)
-		}
+		select {
+		case b := <-m.pool.Full:
+			msg, err := m.parser.Parse(b.Bytes())
+			// Log all message parsing errors.
+			if err != nil {
```

Comment:
Instead of logging all messages, we log only the errors. Libraries shouldn't log on their own, though. They should just provide some means to fetch the errors through return values or on a channel, depending on the code.